simulation/agents.py

```diff
@@ -110,6 +125,48 @@ def renovation_budget(self) -> float:
             self.wealth_percentile,
         )
 
+    @property
+    def insulation_segment(self) -> InsulationSegment:
```

Comment:
I'm on the fence about asking for tests with this. On the one hand, it's from the BEIS report (can you link it to it?). There's not really any behaviour - it's just mapping.

On the other, it's not a very nice bunch of if statements and I worry about it being unintentionally broken in the future. The behaviours in the get quote method rely on this property working. Maybe a sensible compromise is a test like `test_new_roof_is_cheaper_for_small_bungalow_than_large_detatched`. I'm open to discussion...

Reply:
Yeah, it's a bit of limbo. It's not ingested from source via the data warehouse, so it has bypassed some of the DQ/validation at that layer. On the other hand, I don't think it really feel it makes sense to devote too much time to such tests as a) I think we'd need to devote disproportionate effort and a large N of permutations to validate here, and b) I think major errors would likely surface out at the post-simulation analysis of our baseline scenario (e.g. if observing that large flats pay less for insulation than small flats).

Reply:
Ok. I understand and agree about all the permutations. My preference and recommendation would be to have a couple simple cases, but if you want to leave it for post-simulation analysis, I'm ok with that.